Updated Doc Strings

zytegalaxy opened the pull request:
Took a first pass at updating doc strings. README still needs to be updated and set of demos and examples should be added to explain the use cases for offered widgets.

sassanh committed:
refactor: make constants globally accessible for all kv files with `UBO_GUI_` prefix

zytegalaxy committed:
doc: adding some docstrings to slider

zytegalaxy committed:
docs: updated doc strings

sassanh commented on ubo_gui/__init__.py:
```diff
@@ -1,3 +1,7 @@
+"""
+This module initializes the ubo_gui package and registers various widgets with the Kivy Factory.
```
```suggestion
This module initializes the `ubo_gui` package and registers various widgets with the Kivy Factory.
```

We need to put all code-related/machine-formatted elements in backtick, it includes all variable names, function names, module names, package names, programming language keywords, paths, repository names. Modern code editors all render these backtick elements correctly as code (like in how they are rendered in markdown) like this:

<img width="1240" alt="image" src="https://github.com/ubopod/ubo-gui/assets/1270688/c144b2fe-6b67-4449-bd1a-372604bdabc7">

sassanh committed:
refactor: make prompt widget responsive to changes of its items

sassanh committed:
feat: make `MenuWidget` subscribe to the items of the current menu if it provides a `subscribe` property

zytegalaxy committed:
docs: adding back ticks and other improvements

zytegalaxy committed:
docs: improving documentations

zytegalaxy commented:
@sassanh I had to do a rebase which I don't know if It was a good idea. I updated the docs and added backticks. We can further improve the docs but perhaps we can get this PR merged...

sassanh commented:
@mehrdadfeller I checked some files, it looks good. A full review would be very time consuming and considering our priorities, I can't manage to do it. I say lets merge it and then improve it over time if we find any issues.

sassanh commented on ubo_gui/animated_slider/__init__.py:
```diff
@@ -1,3 +1,5 @@
+"""Module for the AnimatedSlider class."""
```
```suggestion
"""Module for the `AnimatedSlider` class."""
```

You missed this one.

mehrdadfeller committed:
Update ubo_gui/animated_slider/__init__.py

Co-authored-by: Sassan Haradji <me@sassanh.com>

mehrdadfeller commented:
@sassanh no merge conflict :) all changes were on docstrings and no heavy review was needed. just wanted to get your feedback before merge
